### PR TITLE
Fixes ARM/Thumb `movt` semantics

### PR DESCRIPTION
**Pull request by bmourad01**
The lifter was incorrectly re-using the upper 16 bits of the destination register, when the manual states that it is completely overwritten by the source operand:

"F5.1.114 MOVT
Move Top writes an immediate value to the top halfword of the destination register. It does not affect the contents
of the bottom halfword."

```
if ConditionPassed() then
EncodingSpecificOperations();
R[d]<31:16> = imm16;
// R[d]<15:0> unchanged
```

So, the previous result of lifting, say, `movt r0, #28` would be:

`R0 := R0 | 0x1C0000`

This PR changes the behavior to be:

`R0 := pad:32[low:16[R0]] | 0x1C0000`

### CODE REVIEW

File: lib/arm/arm_lifter.ml [hunk 1]
```diff
@@ -227,7 +227,8 @@ let lift_move ~encoding mem ops (insn : move_insn) : stmt list =
 
   | `MOVTi16, [|`Reg dest; _; src; cond; _wflag|] ->
     let dest = Env.of_reg dest in
-    [Bil.move dest Bil.(var dest lor exp_of_op src lsl int32 16)] |>
+    let src16 = Bil.(cast unsigned 32 (cast low 16 (var dest))) in
+    [Bil.move dest Bil.(src16 lor exp_of_op src lsl int32 16)] |>
```
Comment (ivg) [inline lib/arm/arm_lifter.ml:231]: It is better not to abuse casts for bit arithmetic. Casts should be used only to change the size of the word. It is much easier to analyze bit-wise operations (especially for SMT solvers) than casts, so my suggestion is to encode it as,

```
movt r0, #0x1c
{
  R0 := R0 & 0xFFFF | 0x1C0000
}
```
i.e.,

```suggestion
    let dest = Env.of_reg dest and src = exp_of_op src in
    Bil.[dest := var dest land int32 0xFFFF lor src lsl int32 16] |>
```

**Review by ivg — CHANGES_REQUESTED** [review]
Thanks for finding it, I am surprised that this bug survived for so many years! 

The only thing is that it is better to use the good old bit operations and encoded it as 
```
rd := rd & 0xffff | (rn << 16)
```
see the suggestion in the review!

File: lib/arm/arm_lifter.ml (hunk 1 above)
Comment (bmourad01) [reply lib/arm/arm_lifter.ml:231]: 👍

File: lib/arm/arm_lifter.ml [hunk 2]
```diff
@@ -227,7 +227,8 @@ let lift_move ~encoding mem ops (insn : move_insn) : stmt list =
 
   | `MOVTi16, [|`Reg dest; _; src; cond; _wflag|] ->
     let dest = Env.of_reg dest in
```
Comment (ivg) [inline lib/arm/arm_lifter.ml:229]: It's hard to code in the browser :D 
```suggestion
```

**Review by ivg — CHANGES_REQUESTED** [review]

**Review by ivg — APPROVED** [review]

**Pull request merged**